src/app/flight-booking/components/flight-result-item/flight-result-item.component.ts

```
@@ -0,0 +1,43 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Flight} from '../../models/flight';
+import {min} from 'rxjs/operators';
+
+@Component({
+  selector: 'app-flight-result-item',
+  templateUrl: './flight-result-item.component.html',
+  styleUrls: ['./flight-result-item.component.scss']
+})
+export class FlightResultItemComponent implements OnInit {
+
+  @Input() flight: Flight;
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+
+  calculateTimeBetween(): string {
+    const legs = this.flight.getLegs();
+    const departureTime = legs[0].departTime;
+    const arrivalTime = legs.length === 2 ? legs[1].arrivalTime : legs.length === 1 ? legs[0].arrivalTime : undefined;
+    if (arrivalTime === undefined) {
+      return 'und';
```

throw an error instead of return